File: Leetcode/2130 Maximum Twin Sum of a Linked List.ts
function pairSum(head: ListNode | null): number {
    let prev: ListNode | null = null;
    let slow = head, fast = head;
    while (fast !== null && fast.next !== null) {
        const _slow = slow as ListNode;
        fast = fast.next.next;
        const nextSlow = _slow.next;
        _slow.next = prev;
        prev = _slow;
        slow = nextSlow;
    }
    let maxPairSum = -Infinity;
    while (prev !== null && slow !== null) {
        maxPairSum = Math.max(maxPairSum, prev.val + slow.val);
        prev = prev.next;
        slow = slow.next;
    }
    return maxPairSum;
}

File: Leetcode/2130 Maximum Twin Sum of a Linked List.py
class Solution:
    def pairSum(self, head):
        prev, slow, fast = None, head, head
        while fast and fast.next:
            fast = fast.next.next
            next_slow = slow.next
            slow.next = prev
            prev = slow
            slow = next_slow
        max_sum = float('-inf')
        while prev and slow:
            max_sum = max(max_sum, prev.val + slow.val)
            prev = prev.next
            slow = slow.next
        return max_sum
